### DISCUSSION TITLE
Using latched buttons for the IncrementDecrementSelector

### DISCUSSION THREAD

**Discussion by ziplob123** [tttapa/Control-Surface#387] [Q&A]
Hi, can anyone give a little advice on how to correct my code so that when I change my CCSmartPot bank I dont have to press the increment and decrement switch twice.
Here is the code I'm using-
************************************************************************************************
#include <Control_Surface.h>
HairlessMIDI_Interface stream;
CCButton But1 = { 8, 62};
CCButton But2 = { 4, 63};
CCButton But3 = { 5, 64};
CCButton But5 = { 6, 65};
CCButton But6 = { 2, 66};
CCButton But8 = { 3, 67};
CCButton But9 = { 10, 68};
CCButton But10 = { 11, 69};

Bank<4> bank;
IncrementDecrementSelector<4> selector = {bank, {7, 9}};

using  CCSmartPot = Bankable::CCSmartPotentiometer<4>;

CCSmartPot potentiometer = {
  {bank, BankType::CHANGE_CHANNEL},     // Bank configuration
  A0,                                   // Analog pin connected to potentiometer
  {MIDI_CC::Channel_Volume, CHANNEL_1}, // Channel volume of channel 1
};

void setup() {
Control_Surface.begin();
}
void loop() 
{
Control_Surface.loop();
}
********************************************************************************************

So Pin 7 and 9 are my bank up and down switches but I have guitar stomp buttons that control my Uno, I think they latch on and off so I have to press them twice to advance up and down the bank or 8 times to do a full cycle through the 4 banks. Can I alter my code in the selector to suit these types of buttons somehow?
Thanks
Ziplob

**Comment by tttapa** [comment]
You'll have to implement your own selector. Have a look at the `GenericIncrementDecrementSelector` and `GenericIncrementSelector` classes for some inspiration:

https://github.com/tttapa/Control-Surface/blob/0efe4a882766894e4e82d5736e4927e92b9d7f41/src/Selectors/IncrementDecrementSelector.hpp#L8-L51

https://github.com/tttapa/Control-Surface/blob/0efe4a882766894e4e82d5736e4927e92b9d7f41/src/Selectors/IncrementSelector.hpp#L8-L47

You'll probably want to use the `Button` class, see the [Button.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/d0/da9/2_8Button_8ino-example.html) example.

**Reply by ziplob123** [reply]
I will have a read and try and get my head around it. I'm still very noobie at coding and it takes a while. Thanks for pointing in the right direction.

**Reply by tmanning47** [reply]
Did you ever make any progress with this? I'm looking to do something similar.

**Reply by ziplob123** [reply]
No mate sorry, I looked at it for a while but it was all a little too much for my coding skills tbh, so I will just buy some non latching switches I think.

**Reply by tmanning47** [reply]
FYI I got a 15 button pedalboard working with a potentiometer to control the banks, just in case it helps you here's the code / discussion: https://github.com/tttapa/Control-Surface/discussions/397#discussioncomment-431570

**Reply by ziplob123** [reply]
Tyvm for sharing.
